### PR TITLE
Table decoder

### PR DESCRIPTION
**Pull request by dereulenspiegel**
I had some trouble getting reliable results with a noisy mechanical rotary decoder, so I started reading into this issue and finally found https://www.best-microcontroller-projects.com/rotary-encoder.html which showed how table based decoding can be used to get more reliability on noisy encoders.
The C version seems to work fine, this rust port needs real world testing though.
With this PR I would like to check if this change within the intended scope of the library and if such changes are welcome.

### CODE REVIEW

**Comment by leshow** [comment]
Yes, I have seen this article actually. Thankfully I didn't need it for the encoders I've got around here, but if it helps you I'm happy to consider merging something like this in.

File: Cargo.toml [hunk 1]
```diff
@@ -10,15 +10,19 @@ license = "MIT"
...
-either = { version = "1.5.3", default-features = false }
+embedded-hal = { version = "0.2.4", features = ["unproven"] }
+either = { version = "1.6.1", default-features = false }
 
 [dev-dependencies]
-version-sync = "0.8"
-cortex-m = "0.6.1"
-cortex-m-rt = "0.6.10"
+version-sync = "0.9.1"
+cortex-m = "0.6.3"
```
Comment (leshow) [inline Cargo.toml]: Shall we just go with `0.6` for both? I'm not sure why I specified patch version numbers oiriginally, it's pretty unnecessary. Same with the `embedded-hal` and `either` versions if you wouldn't mind

File: src/lib.rs [hunk 2]
```diff
@@ -73,4 +86,38 @@ where
...
     }
+
+
+    #[cfg(feature = "table-decoder")]
+    /// Call `update` to evaluate the next state of the encoder, propagates errors from `InputPin` read
+    pub fn update(&mut self) -> Result<Direction, Either<A::Error, B::Error>> {
+        // Implemented after https://www.best-microcontroller-projects.com/rotary-encoder.html
+        self.prev_next <<= 2;
+        if self.pin_a.is_high().map_err(Either::Left)? {
+            self.prev_next |= 0x02;
```
Comment (leshow) [inline src/lib.rs:107]: I find the binary a lot easier to read and figure out what's going on, can we write these as `0b0010` and `0b0001`?

edit: nevermind, I wouldn't want to have to write out like `0b1111_1111` below, we can just leave it the way it is.

File: Cargo.toml (hunk 1 above)
Comment (dereulenspiegel) [reply Cargo.toml]: Agree. I will push this with the other changes into this branch.

**Comment by dereulenspiegel** [comment]
I did some cleanup, tested everything with a very cheap and noisy mechanical rotary encoder on a blue pill and it seems to be working fine.
I would like to add unit tests for this, but I am not sure how to simulate noisyness etc.

**Comment by leshow** [comment]
Yeah good idea. I think we can still merge this though. We can both think
about what the test may look like and submit a future PR for it. I won't be
cutting a release immediately anyway.

**Comment by leshow** [comment]
Thanks so much for the contribution!

**Pull request merged**